lib/Controller/SocialApiController.php

```diff
@@ -0,0 +1,300 @@
+
+		$supported = array();
+		$supported['avatar'] = array();
+
+		foreach(array_keys(self::SOCIAL_CONNECTORS) as $network) {
+			array_push($supported['avatar'], $network);
+		}
+
+		if (strcmp($type, 'all') === 0) {
+			// return array of arrays
+			return $supported;
+		}
+		if (array_key_exists($type, $supported)) {
+			return $supported[$type];
+		}
+		// unknown type
+		return array();
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * generate download url for a social entry (based on type of data requested)
+	 *
+	 * @param {array} socialentries the network and id from the social profiles
+	 * @param {array} network the choice which network to use or 'first' to use any
+	 * @returns {String} the url to the requested information or null in case of errors
+	 */
+	protected function getSocialConnector(array $socialentries, string $network) : ?string {
+
+		$connector = null;
+		$selection = array();
+
+		// get all supported networks
+		if ($network === 'first') {
+			$selection = self::SOCIAL_CONNECTORS;
+		} else {
+			$selection = array($network => self::SOCIAL_CONNECTORS[$network]);
+		}
+
+		// check selected networks in order
+		foreach($selection as $socialnet => $social) {
+
+			// search for this network in user's profile
+			foreach ($socialentries as $socialnetentry => $profileId) {
+
+				if ($socialnet === strtolower($socialnetentry)) {
+					// cleanups
+					if (in_array('basename', $social['cleanups'])) {
+						$profileId = basename($profileId);
+					}
+					if (in_array('filter', $social['cleanups'])) {
+						if (preg_match($social['filter']['regex'], $profileId, $matches)) {
+							$profileId = $matches[$social['filter']['group']];
+						}
+					}
+					// checks
+					if (in_array('number', $social['checks'])) {
+						if (!ctype_digit($profileId)) {
+							break;
+						}
+					}
+					$connector = str_replace("{socialId}", $profileId, $social['recipe']);
+					break;
+				}
+			}
+			if ($connector) {
+				break;
+			}
+		}
+		return ($connector);
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * Retrieves social profile data for a contact
+	 *
+	 * @param {String} addressbookId the addressbook identifier
+	 * @param {String} contactId the contact identifier
+	 * @param {String} type the kind of information to retrieve -- provision
+	 * @param {String} network the social network to use or 'first' to use any
+	 *
+	 * @returns {JSONResponse} an empty JSONResponse with respective http status code
+	 */
+	public function fetch(string $addressbookId, string $contactId, string $type, string $network) : JSONResponse {
+
+		$url = null;
+
+		try {
+			// get corresponding addressbook
+			$addressBooks = $this->manager->getUserAddressBooks();
+			$addressBook = null;
+			foreach($addressBooks as $ab) {
+				if ($ab->getUri() === $addressbookId) {
+					$addressBook = $ab;
+				}
+			}
+			if (is_null($addressBook)) {
+				return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
```

Comment:
I see a lot of STATUS_INTERNAL_SERVER_ERROR.
This is not correct, the server is behaving properly.
The request is a bad one :)

Reply:
good point, I changed it.